prime-router/src/test/kotlin/azure/SettingsFunctionTests.kt

```diff
@@ -0,0 +1,581 @@
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        var response = settingsFunction.getReceivers(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.getReceivers(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.getOneReceiver(
+            httpRequestMessage,
+            "$organizationName.elr-secondary",
+            "elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.getOneReceiver(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2",
+            "test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `test access user cannot update their organization's receivers (updateOneReceiver)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting(HttpMethod.PUT)
+        val response = settingsFunction.updateOneReceiver(
+            httpRequestMessage,
+            "$organizationName.elr-secondary",
+            "elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access user cannot update another organization's receivers (updateOneReceiver)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting(HttpMethod.PUT)
+        val response = settingsFunction.updateOneReceiver(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2",
+            "test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access DHPrimeAdmins can update any organization's receivers (updateOneReceiver)`() {
+        val settingsFunction = setupSettingsFunctionForTesting("DHPrimeAdmins", mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting(HttpMethod.PUT)
+        var response = settingsFunction.updateOneReceiver(
+            httpRequestMessage,
+            organizationName,
+            "elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.updateOneReceiver(
+            httpRequestMessage,
+            otherOrganizationName,
+            "test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `test access user cannot get their organization's revision history (getSettingRevisionHistory)`() {
```

Comment:
Same issue here, not passing the organization

Reply:
fixed